refactor(no-drivable-lane): use getOrDeclareParameter

TakaHoribe opened the pull request:
## Description

use `tier4_autoware_utils::getOrDeclareParameter` for the ros parameter to prevent duplicate parameter declaration.

## Tests performed

Run psim

## Effects on system behavior

None

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [x] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

TakaHoribe commented on planning/behavior_velocity_no_drivable_lane_module/src/manager.cpp:
```diff
@@ -14,17 +14,20 @@
   const std::string ns(getModuleName());
-  planner_param_.stop_margin = node.declare_parameter(ns + ".stop_margin", 1.5);
-  planner_param_.print_debug_info = node.declare_parameter(ns + ".print_debug_info", false);
+  planner_param_.stop_margin = getOrDeclareParameter<double>(node, ns + ".stop_margin");
```
```suggestion
  planner_param_.stop_margin = getOrDeclareParameter<double>(node, ns + ".stop_margin", 1.5);
```

TakaHoribe commented on planning/behavior_velocity_no_drivable_lane_module/src/manager.cpp:
```diff
@@ -14,17 +14,20 @@
-  planner_param_.stop_margin = node.declare_parameter(ns + ".stop_margin", 1.5);
-  planner_param_.print_debug_info = node.declare_parameter(ns + ".print_debug_info", false);
+  planner_param_.stop_margin = getOrDeclareParameter<double>(node, ns + ".stop_margin");
+  planner_param_.print_debug_info = getOrDeclareParameter<bool>(node, ns + ".print_debug_info");
```
```suggestion
  planner_param_.print_debug_info = getOrDeclareParameter<bool>(node, ns + ".print_debug_info, false");
```

TakaHoribe commented on planning/behavior_velocity_no_drivable_lane_module/src/manager.cpp:
```diff
@@ -14,17 +14,20 @@
-  planner_param_.stop_margin = node.declare_parameter(ns + ".stop_margin", 1.5);
-  planner_param_.print_debug_info = node.declare_parameter(ns + ".print_debug_info", false);
+  planner_param_.stop_margin = getOrDeclareParameter<double>(node, ns + ".stop_margin", 1.5);
+  planner_param_.print_debug_info = getOrDeclareParameter<bool>(node, ns + ".print_debug_info, false");
```
```suggestion
  planner_param_.print_debug_info = getOrDeclareParameter<bool>(node, ns + ".print_debug_info", false);
```

satoshi-ota commented on planning/behavior_velocity_no_drivable_lane_module/src/manager.cpp:
```diff
@@ -14,17 +14,20 @@
   const std::string ns(getModuleName());
-  planner_param_.stop_margin = node.declare_parameter(ns + ".stop_margin", 1.5);
-  planner_param_.print_debug_info = node.declare_parameter(ns + ".print_debug_info", false);
+  planner_param_.stop_margin = getOrDeclareParameter<double>(node, ns + ".stop_margin", 1.5);
```
```suggestion
  planner_param_.stop_margin = getOrDeclareParameter<double>(node, ns + ".stop_margin");
```

satoshi-ota commented on planning/behavior_velocity_no_drivable_lane_module/src/manager.cpp:
```diff
@@ -14,17 +14,20 @@
-  planner_param_.stop_margin = node.declare_parameter(ns + ".stop_margin", 1.5);
-  planner_param_.print_debug_info = node.declare_parameter(ns + ".print_debug_info", false);
+  planner_param_.stop_margin = getOrDeclareParameter<double>(node, ns + ".stop_margin", 1.5);
+  planner_param_.print_debug_info = getOrDeclareParameter<bool>(node, ns + ".print_debug_info", false);
```
```suggestion
  planner_param_.print_debug_info = getOrDeclareParameter<bool>(node, ns + ".print_debug_info");
```

ahmeddesokyebrahim commented:
Thank you @TakaHoribe san for this PR.
Actually I am trying to checkout this branch to test it in one of `no_drivable_lane` scenarios and I am not able to compile.
As well, there is some required gityub checks failing.
Can you please rebase and have a look to these checks ?
Thanks again for your efforts.

TakaHoribe commented:
@ahmeddesokyebrahim Thank you for your review. I fixed the PR and also added a missing parameter in the node test. It should be fine I hope. Please check after the CI passes.

TakaHoribe commented:
@ahmeddesokyebrahim Oops. Sorry, since I enabled an auto-merge, now it is merged!
